metashare/api/migrations/0030_replace_repo_url_with_repo_id.py

```diff
@@ -0,0 +1,94 @@
+# Generated by Django 2.2.6 on 2019-10-15 18:41
+
+from urllib.parse import urlparse
+
+from django.db import migrations, models
+
+import sfdo_template_helpers.fields.string
+
+from ..gh import get_repo_info
+
+
+def extract_owner_and_name(gh_url):
+    path = urlparse(gh_url).path
+    _, owner, name, *_ = path.split("/")
+    return owner, name
+
+
+def set_repo_owner_and_name(apps, schema_editor):
+    Repository = apps.get_model("api", "Repository")
+    for instance in Repository.objects.all():
+        owner, name = extract_owner_and_name(instance.repo_url)
+        if not instance.repo_name:
+            instance.repo_name = name
+        if not instance.repo_owner:
+            instance.repo_owner = owner
+        instance.save()
+
+
+def set_repo_url(apps, schema_editor):
+    Repository = apps.get_model("api", "Repository")
+    for instance in Repository.objects.all():
+        owner = instance.repo_owner
+        name = instance.repo_name
+        instance.repo_url = f"https://github.com/{owner}/{name}"
+        instance.save()
+
+
+def set_repo_id(apps, schema_editor):
+    GitHubRepository = apps.get_model("api", "GitHubRepository")
+    for instance in GitHubRepository.objects.all():
+        if not instance.repo_id:
+            owner, name = extract_owner_and_name(instance.repo_url)
+            repo = get_repo_info(instance.user, repo_owner=owner, repo_name=name)
```

Comment:
I opted to lazily fill repo ID because I didn't think we could rely on having valid GitHub tokens. If you feel confident we will always have that when this migration is run, I'm OK with it, but I'm reticent.

Reply:
Ah, that's a fair point. I'd say since GH repos are easy to repopulate, we could either just remove them all in this migration (which should trigger a re-sync on next login/page-load), or move this into a try/catch and delete ones for which we don't currently have access.